### PR TITLE
refactored classlist functions to support multiple class names

### LINKED ISSUE #306: CSS class-list should handle more than one class
`add`, `has`, `toogle` and `remove` should be able to deal with mora than just a single class.

### PR DESCRIPTION
**Pull request by AndyOGo**
Fixes #306 .

Changes proposed in this pull request:

 - added support for multiple classes
 - needs to refactor existing code

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update
 
 # How Has This Been Tested?
 
locally
 
 # Checklist:
 
all done

### CODE REVIEW

File: src/components/m-header-navigation/js/stroke.js [hunk 1]
```diff
@@ -87,8 +87,7 @@ class Stroke extends UiEvents {
     if (has(node, this.options.activeClass)) {
       this._activeNode = node;
 
-      add(node, this.options.activeOpenClass);
-      add(node, this.options.activeMoveClass);
+      add(node, this.options.activeOpenClass, this.options.activeMoveClass);
```
Comment (AndyOGo) [inline src/components/m-header-navigation/js/stroke.js]: @LucaMele 
@TheDadi 
Is it really worth this update?
I checked all other components, only the stroke needs to `add`/`remove` multiple classes at once on the same DOM node.
Valid Reason for me are:
- Pattern-Library consumers won't need to worry if they need to attach multiple classes
- It is more coerce the `Element.classList` API https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

Comment (AndyOGo) [reply src/components/m-header-navigation/js/stroke.js]: @LucaMele 
@TheDadi 
Knock knock

Comment (LucaMele) [reply src/components/m-header-navigation/js/stroke.js]: :)

File: src/js/class-list.js [hunk 2]
```diff
@@ -12,56 +12,142 @@ function getReClass(className, modifier) {
...
-  if (!has(node, className)) {
-    node.className += ` ${className}`;
+export function add(node, ...classNames) {
+  const hasResult = has(node, ...classNames);
+
+  if (hasResult === true) {
+    return;
+  }
+
+  let classesToAdd = [...classNames];
```
Comment (LucaMele) [inline src/js/class-list.js]: u might want to cache deconstruction before u do has(node, ...classNames)

File: src/js/class-list.js [hunk 3]
```diff
@@ -12,56 +12,142 @@ function getReClass(className, modifier) {
...
+  const { className: nodeClassName } = node;
+  let hasAtLeastOne = false;
+  let hasAllClasses = true;
 
-  return regexClass.test(node.className);
+  const hasArray = classNames.map(mapHas);
+
+  return hasAllClasses || hasAtLeastOne ? hasArray : false;
+
+  function mapHas(className) {
```
Comment (LucaMele) [inline src/js/class-list.js]: grrrr hoisting

File: src/js/class-list.js [hunk 4]
```diff
@@ -12,56 +12,142 @@ function getReClass(className, modifier) {
...
-    const regexClass = getReClass(className, 'g');
+export function remove(node, ...classNames) {
+  const hasResult = has(node, ...classNames);
 
-    node.className = node.className.replace(regexClass, ' ').replace(regexTrim, ' ');
+  if (hasResult === false) {
+    return;
   }
+
+  let classesToRemove = [...classNames];
```
Comment (LucaMele) [inline src/js/class-list.js]: dtto

**Pull request merged**